homeassistant/components/binary_sensor/hue.py

```diff
@@ -0,0 +1,338 @@
+    # when we update, we know all commands have been processed. If we trigger
+    # the update from inside async_turn_on, the update will not capture the
+    # changes to the second entity until the next polling update because the
+    # throttle decorator will prevent the call.
+
+    progress = None
+    sensor_progress = set()
+
+    async def request_update(object_id):
+        """Request an update.
+
+        We will only make 1 request to the server for updating at a time. If a
+        request is in progress, we will join the request that is in progress.
+
+        This approach is possible because should_poll=True. That means that
+        Home Assistant will ask sensors for updates during a polling cycle or
+        after it has called a service.
+
+        We keep track of the sensors that are waiting for the request to finish.
+        When new data comes in, we'll trigger an update for all non-waiting
+        sensors. This covers the case where a service is called to modify 2
+        sensors but in the meanwhile some other sensor has changed too.
+        """
+        nonlocal progress
+
+        progress_set = sensor_progress
+        progress_set.add(object_id)
+
+        if progress is not None:
+            return await progress
+
+        progress = asyncio.ensure_future(update_bridge())
+        result = await progress
+        progress = None
+        sensor_progress.clear()
+        return result
+
+    async def update_bridge():
+        """Update the values of the bridge.
+
+        Will update sensors from the bridge.
+        """
+        tasks = []
+        tasks.append(async_update_items(
+            hass, bridge, async_add_devices, request_update,
+            cur_sensors, sensor_progress
+        ))
+
+        await asyncio.wait(tasks)
+
+    await update_bridge()
+
+
+async def async_update_items(hass, bridge, async_add_devices,
+                             request_bridge_update, current,
+                             progress_waiting):
+    """Update sensors from the bridge."""
+    import aiohue
+    allow_clip_sensors = bridge.allow_clip_sensors
+
+    api = bridge.api.sensors
+
+    try:
+        with async_timeout.timeout(4):
+            await api.update()
+    except (asyncio.TimeoutError, aiohue.AiohueException):
+        if not bridge.available:
+            return
+
+        _LOGGER.error('Unable to reach bridge %s', bridge.host)
+        bridge.available = False
+
+        for sensor_id, sensor in current.items():
+            if sensor_id not in progress_waiting:
+                sensor.async_schedule_update_ha_state()
+
+        return
+
+    if not bridge.available:
+        _LOGGER.info('Reconnected to bridge %s', bridge.host)
+        bridge.available = True
+
+    new_sensors = []
+
+    if allow_clip_sensors:
+        allowed_binary_sensor_types = ALL_BINARY_SENSORS
+    else:
+        allowed_binary_sensor_types = HUE_BINARY_SENSORS
+
+    for item_id in api:
```

Comment:
```py
for item_id, sensor in api.items():
```

Reply:
I'm new to python and I tried your construction instead of what I used but yours returns:
```
AttributeError: 'Sensors' object has no attribute 'items'
```